shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1279,6 +1280,7 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
   [self invalidateDisplayLink];
 
   self.displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(onDisplayLink)];
+  [DisplayLinkManager setMaxRefreshRateIfEnabled:self.displayLink];
```

Comment:
Can we instead use `VSyncClient` for `self.displayLink` so the displayLink used for keyboard will be in sync with the main displayLink?

You don't need to expose `setMaxRefreshRateIfEnabled:self` as a class method if doing above.

Reply:
Hi @cyanglaz, But I need a continuous vsync callback `See FlutterViewController.onDisplayLink` to update the keyboard properties until the keyboard animation is done.

Reply:
And properties in `FlutterViewController.onDisplaylink` must run in iOS main thread. But the displaylink in `VsyncClient` is run in `UITaskRunner` which is not in iOS main thread.

Reply:
VsyncClient can be used on any task runner: https://github.com/flutter/engine/blob/main/shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.h#L33

You can initialize it using a PlatformTaskRunner.

Reply:
Oh, I got what you mean now. (I thought you said that use the same vsync_client object in UITaskRunner before😄)......I will try to get a new displaylink by createing a new `VsyncClient` in `FlutterViewController` to do this. Thx @cyanglaz